lib/inspec/resource.rb

```diff
@@ -25,9 +27,21 @@ def self.new_registry
     #
     # @param backend [BackendRunner] exposing the target to resources
     # @return [ResourcesDSL]
-    def self.create_dsl(backend, my_registry = registry)
-      # need the local name, to use it in the module creation further down
+    def self.create_dsl(profile_context)
+      backend = profile_context.backend
+      my_registry = profile_context.resource_registry
+
       Module.new do
+        define_method :resource_class do |profile_name, resource_name|
+          inner_context = profile_context.subcontext_by_name(profile_name)
+          fail ProfileNotFound, "Cannot find profile named: #{profile_name}" if inner_context.nil?
+          inner_context.resource_registry[resource_name]
+        end
+
+        define_method :resource do |profile_name, resource_name, *args|
```

Comment:
That depends.  Do we want a means of accessing specific resources without mutating the current namespace or not?  If no, then we can certainly remove this.